code/datums/ai/monkey/monkey_subtrees.dm

```diff
@@ -1,102 +1,63 @@
-				return SUBTREE_RETURN_FINISH_PLANNING //Focus on this
-
-			else //He's down, can we disposal him?
-				var/obj/machinery/disposal/bodyDisposal = locate(/obj/machinery/disposal/) in view(MONKEY_ENEMY_VISION, living_pawn)
-				if(bodyDisposal)
-					controller.blackboard[BB_MONKEY_CURRENT_ATTACK_TARGET] = selected_enemy
-					controller.blackboard[BB_MONKEY_TARGET_DISPOSAL] = bodyDisposal
-					controller.queue_behavior(/datum/ai_behavior/disposal_mob, BB_MONKEY_CURRENT_ATTACK_TARGET, BB_MONKEY_TARGET_DISPOSAL)
-					return SUBTREE_RETURN_FINISH_PLANNING
+	if(prob(50))
+		controller.queue_behavior(/datum/ai_behavior/use_on_object, BB_MONKEY_CURRENT_PRESS_TARGET)
+		return SUBTREE_RETURN_FINISH_PLANNING
+
+	if(!controller.blackboard[BB_MONKEY_CURRENT_GIVE_TARGET])
+		controller.queue_behavior(/datum/ai_behavior/find_and_set/pawn_must_hold_item, BB_MONKEY_CURRENT_GIVE_TARGET, /mob/living, 2)
+		return
 
 	if(prob(5))
-		controller.queue_behavior(/datum/ai_behavior/use_in_hand)
+		controller.queue_behavior(/datum/ai_behavior/give, BB_MONKEY_CURRENT_GIVE_TARGET)
+		return SUBTREE_RETURN_FINISH_PLANNING
 
-	if(selected_enemy || !DT_PROB(MONKEY_SHENANIGAN_PROB, delta_time))
+	controller.TryFindWeapon()
+
+///monkey combat subtree.
+/datum/ai_planning_subtree/monkey_combat/SelectBehaviors(datum/ai_controller/monkey/controller, delta_time)
+	var/mob/living/living_pawn = controller.pawn
+	var/list/enemies = controller.blackboard[BB_MONKEY_ENEMIES]
+
+	if((HAS_TRAIT(controller.pawn, TRAIT_PACIFISM)) || (!length(enemies) && !controller.blackboard[BB_MONKEY_AGGRESSIVE])) //Pacifist, or we have no enemies and we're not pissed
 		return
 
-	if(world.time >= controller.blackboard[BB_MONKEY_NEXT_HUNGRY])
-		var/list/food_candidates = list()
-		for(var/obj/item as anything in living_pawn.held_items)
-			if(!item || !controller.IsEdible(item))
-				continue
-			food_candidates += item
+	if(!controller.blackboard[BB_MONKEY_CURRENT_ATTACK_TARGET])
+		controller.queue_behavior(/datum/ai_behavior/monkey_set_combat_target, BB_MONKEY_CURRENT_ATTACK_TARGET, BB_MONKEY_ENEMIES)
+		return SUBTREE_RETURN_FINISH_PLANNING
 
-		for(var/obj/item/candidate in oview(2, living_pawn))
-			if(!controller.IsEdible(candidate))
-				continue
-			food_candidates += candidate
+	var/datum/weakref/target_ref = controller.blackboard[BB_MONKEY_CURRENT_ATTACK_TARGET]
+	var/mob/living/selected_enemy = target_ref?.resolve()
 
-		if(length(food_candidates))
-			var/obj/item/best_held = controller.GetBestWeapon(null, living_pawn.held_items)
-			for(var/obj/item/held as anything in living_pawn.held_items)
-				if(!held || held == best_held)
-					continue
-				living_pawn.dropItemToGround(held)
+	if(!selected_enemy.stat) //He's up, get him!
+		if(living_pawn.health < MONKEY_FLEE_HEALTH) //Time to skeddadle
+			controller.queue_behavior(/datum/ai_behavior/monkey_flee)
+			return SUBTREE_RETURN_FINISH_PLANNING //I'm running fuck you guys
```

Comment:
I'd like a check here to nullify the current target if the target doesn't exist.
I have a reproducible repeating runtime error on line 42 if their target stops existing.

Reply:
I believe I fixed the runtime